doc/production-hosting.md

```diff
@@ -147,64 +116,25 @@ hosted on [Zeit Now][]. It's managed in the
 
 ## Deployment
 
-To set things up for deployment:
-
-1.  Get your SSH key added to the server.
-2.  Clone a fresh copy of the repository, dedicated for deployment.
-    (Not required, but recommended; and lets you use `npm ci` below.)
-3.  Add remotes:
-
-```sh
-git remote add s0 root@s0.servers.shields.io:/home/m/shields.git
-git remote add s1 root@s1.servers.shields.io:/home/m/shields.git
-git remote add s2 root@s2.servers.shields.io:/home/m/shields.git
-```
-
-`origin` should point to GitHub as usual.
-
-4.  Since the deploy uses `git worktree`, make sure you have git 2.5 or later.
-
-To deploy:
+The deployment is done in two stages: the badge server (heroku) and the front-end (gh-pages).
 
-1.  Use `git fetch` to obtain a current copy of
-    `local-shields-io-production.yml` from the server (or obtain the current
-    version of that file some other way). Save it in `config/`.
-2.  Check out the commit you want to deploy.
-3.  Run `npm ci`. **This is super important for the frontend build!**
-4.  Run `make deploy-s0` to make a canary deploy.
-5.  Check the canary deploy:
-    - [Visit the server][s0]. Don't forget that most of the preview badges
-      are static!
-    - Look for errors in [Sentry][].
-    - Keep an eye on the [status page][status].
-6.  After a little while (usually 10–60 minutes), finish the deploy:
-    `make push-s1 push-s2 deploy-gh-pages`.
+### Heroku
 
-To roll back, check out the commit you want to roll back to and repeat those
-steps.
+After merging a commit to master, heroku should create a staging deploy. Check this has deployed correctly in the `shields-staging` pipeline and review http://shields-staging.herokuapp.com/
 
-To see which commit is deployed to a server run `git ls-remote` and then
-`git log` on the `HEAD` ref. There will be two deploy commits preceded by the
-commit which was deployed.
+If we're happy with it, "promote to production". This will deploy what's on staging to the `shields-production-eu` and `shields-production-us` pieplines.
 
-Be careful not to push the deploy commits to GitHub.
+### Frontend
 
-`make deploy-s0` does the following:
+To deploy the front-end to GH pages, use a clean clone of the shields repo.
 
-1.  Creates a working tree in `/tmp`.
-2.  In that tree, runs `features` and `examples` to generate data files
-    needed for the frontend.
-3.  Builds and checks in the built frontend.
-4.  Checks in `local-shields-io-production.yml`.
-5.  Pushes to s0, which updates dependencies and then restarts itself.
-
-`make push-s1 push-s2 deploy-gh-pages` does the following:
+```sh
+$ git pull  # update the working copy
+$ npm install  # install dependencies (devDependencies are needed to build the frontend)
```

Comment:
To respect the lockfile, this should be `npm ci`.

Reply:
`npm install` doesn't ignore the lockfile, but it will update the lockfile if `package.json` doesn't match it. While `npm ci` would exit with a non-zero status code in that case, if you've made uncommited changes to `package.json`, `git pull` would have failed in the previous step, so.. meh :shrug: . I've changed it anyway.

Reply:
If `package.json` specifies e.g. `^1.0.2` and `1.1.0` is available but we haven't upgraded, it will upgrade it and update the lockfile. We want that to happen when we're bumping the dependencies in a PR, not when we're deploying.

Reply:
Its not deeply important for merging this PR, but I'm just trying to understand what you're saying here..

If I create the following `package.json`:

```json
{
  "dependencies": {
    "is-png": "^1.0.0"
  }
}
```

and corresponding `package-lock.json`:

```json
{
  "requires": true,
  "lockfileVersion": 1,
  "dependencies": {
    "is-png": {
      "version": "1.0.0",
      "resolved": "https://registry.npmjs.org/is-png/-/is-png-1.0.0.tgz",
      "integrity": "sha1-PYA3P+m4nWX9NB9lnT/AoRNecYo="
    }
  }
}
```

even though [is-png 1.1.0](https://www.npmjs.com/package/is-png/v/1.1.0) exists, I can `npm install` in that project dir as many times as you like. `package-lock.json` is always going to have `1.0.0` in it, and so is `node_modules`.

Are you saying you think that just doing an `npm install` transparently bumps that to `1.1.0`, or am I misunderstanding a more subtle point here?